### ISSUE TITLE
Flatpak support

### ISSUE THREAD

**Issue by Eonfge** [flacon/flacon#97]
# Intro
Flatpak is a framework for distributing desktop applications on Linux. It has been created by developers who have a long history of working on the Linux desktop, and is run as an independent open source project.

[Introduction: Reasons to use Flatpak](http://docs.flatpak.org/en/latest/introduction.html#reasons-to-use-flatpak)

# Why Flatpak
* Users will always be able to use the latest version, no _ppa_ or _aur_ required
* It works the same on all distributions
* The package can be bundled with all required components in one go 
* The package is not dependent on the architecture of the host machine
* Improved security for the user because Flatpak is sandboxed and controlled by permissionssets

# What is required to do so
* Choose a shared library to work from, likely org.kde.Platform because of KDE
* Compile your dependencies for this package
* Compile Flacon for this package
* Create a Continuous Integration script that can do this for you
* Publish the combined app using Flathub (or other)

**Comment by SokoloffA** [comment]
I have prepared a flatpak package for testing. Please let me know, is it working fine for you. If all is OK, I'll push package to the flathub.

[flacon-5.4.0.flatpak.tar.gz](https://github.com/flacon/flacon/files/3498062/flacon-5.4.0.flatpak.tar.gz)

**Comment by Eonfge** [comment]
I found a few issues:

- There is still a tab in the `preferences` menu which makes no sense for the Flatpak version. All codecs and libs are packaged into the Flatpak and it is security-wise impossible to use alternative versions.

_It would recommend to remove this tab for flatpak builds._

- Second, Setting a temporary directory doesn't work well. It also doesn't make much sense in the Flatpak context, since each app has his own `/var/tmp`.  It gives this error in the console why I run it:
```
(flacon:2): dconf-WARNING **: 19:33:41.129: failed to commit changes to dconf: GDBus.Error:org.freedesktop.DBus.Error.ServiceUnknown: org.freedesktop.DBus.Error.ServiceUnknown
```
_I would propose that you set the tmp path to the Flatpak default, and that you remove the UI element._

**Comment by Eonfge** [comment]
Extra details, I've been testing this with

Ubuntu **19.04**
Flatpak **1.2.4**

I also want to test a .ape file since they have historically caused many problems... But I have no .ape file at the moment, so I'll first rip a CD I have to I can test it later. I'll let you know if my experiment succeeds.

**Comment by Eonfge** [comment]
Converting .ape images is also confirmed to work. Important to know, is that I was previously unable to convert .ape files. This is nice, because this directly mitigates the following issues #94, #90, #74, #70

Considering this major success, I would actually encourage you to stick with Flatpak as a primary means of distribution.

**Comment by Eonfge** [comment]
He, since I reported this issue 3 months ago, I've gained a lot of experience with packaging for Flatpak. If you would like to have some help with publishing Flacon on Flathub, let me know.

**Comment by SokoloffA** [comment]
Thanks for your help. Unfortunately, I don't have no time, so it took so many months. Sorry.

I wrote the [script to build Flatpack](https://github.com/flacon/packages/tree/master/Flatpak). But there were a couple of bugs (#99, #100). I plan to fix them, and make a new release this weekend. The new version will also be available in Flatpack.

If there's a problem, I'll ask for your help. Thanks again!

**Comment by Eonfge** [comment]
Hey, I made a releasable version of 5.4.0:
https://github.com/Eonfge/flathub/tree/com.github.Flacon

It's based on the Flathub repo and with your approval, I would gladly start the review process. I will ask them to make you a maintainer as well, so that you remain in control.

The only changes I made compared to your script, are a few updated runtimes and I made a patch to apply the `-DFLATPAK_BUNDLE=YES` option.
